### ISSUE TITLE
Setting server open to true focuses the tab when config.mts is changed

### ISSUE THREAD

**Issue by gkiely** [vuejs/vitepress#4040]
### Describe the bug

Setting server open to true in the config.mts and changing the config will focus the localhost tab, this is unexpected and differs from how vite behaves normally.

### Reproduction
```ts
import { defineConfig } from 'vitepress';

// https://vitepress.dev/reference/site-config
export default defineConfig({
  title: "Example project",
  description: "An example project",
  themeConfig: {
    // https://vitepress.dev/reference/default-theme-config
    nav: [
      { text: 'Home', link: '/' },
      { text: 'Examples', link: '/markdown-examples' }
    ],

    sidebar: [
      {
        text: 'Examples',
        items: [
          { text: 'Markdown Examples', link: '/markdown-examples' },
          { text: 'Examples', link: '/examples' }
        ]
      }
    ],

    socialLinks: [
      { icon: 'github', link: 'https://github.com/vuejs/vitepress' }
    ]
  },
  vite: {
    server: {
      open: true,
    }
  },
})
```

### Expected behavior

Tab does not focus

### System Info

```Text
System:
    OS: macOS 12.7
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.87 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.12.1 - ~/.volta/tools/image/node/20.12.1/bin/node
    Yarn: 1.22.18 - ~/.volta/tools/image/yarn/1.22.18/bin/yarn
    npm: 10.5.0 - ~/.volta/tools/image/node/20.12.1/bin/npm
    bun: 1.1.18 - ~/code/youneedawiki/node_modules/.bin/bun
    Watchman: 2023.04.10.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 114.1.52.129
    Chrome: 126.0.6478.127
    Edge: 126.0.2592.102
    Safari: 17.5
```

### Additional context
https://www.loom.com/share/c1bc78328066473cae5d856530101559?sid=25891b7f-683f-4d26-a13e-425418fbea95

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Which vitepress version are you using?

**Comment by gkiely** [comment]
> Which vitepress version are you using?

1.3.0

**Comment by brc-dd** [comment]
Looks like vite.open is not supported yet. If you run with `vitepress dev docs --open` it should work fine.

**Comment by brc-dd** [comment]
Release a fix in v1.3.1. Should work fine now.
